comfyui_script_to_video_suite/s2v_nodes/s2v_prompt_gen_node.py

```
@@ -0,0 +1,118 @@
+            }
+        }
+
+    RETURN_TYPES = ("STRING",)
+    RETURN_NAMES = ("final_prompts",)
+    FUNCTION = "generate_prompts_in_batches"
+    CATEGORY = "Script To Video Suite"
+
+    def _split_storyboard_into_panels(self, storyboard_text: str) -> list[str]:
+        """Splits the full storyboard into a list of individual panel strings."""
+        panels = re.split(r'\s*--- PANEL BREAK ---\s*|(?=PANEL\s+\d+)', storyboard_text)
+        cleaned_panels = [p.strip() for p in panels if p.strip() and p.startswith("PANEL")]
+        print(f"✅ Storyboard split into {len(cleaned_panels)} individual panels.")
+        return cleaned_panels
+
+    def generate_prompts_in_batches(self, storyboard_text: str, master_prompt: str, batch_size: int, debug_mode: bool = False, debug_filepath: str = ""):
+        
+        # --- NEW: DEBUG MODE LOGIC ---
+        if debug_mode:
+            print("💡 DEBUG MODE IS ON. Ignoring 'storyboard_text' input and loading from file.")
+            try:
+                if not os.path.exists(debug_filepath):
+                    raise FileNotFoundError(f"Debug file not found at path: {debug_filepath}")
+                
+                with open(debug_filepath, 'r', encoding='utf-8') as f:
+                    storyboard_text = f.read()
+                print(f"✅ Successfully loaded debug data from: {debug_filepath}")
+            except Exception as e:
+                # Raise an exception to make the node turn red in ComfyUI and show the error.
+                error_message = f"❌ DEBUG ERROR: Failed to load file. Reason: {e}"
+                print(error_message)
+                raise e  # Stop execution
+        else:
+            print("Executing 'Prompt Generator' node in normal mode...")
+        
+        if not storyboard_text or storyboard_text.strip() == "":
+            print("⚠️ WARNING: Storyboard text is empty (either from input or debug file). Aborting.")
+            return ("",)
```

silent failure... may be add better error handling